pkg/devfile/parser/parse_test.go

```diff
@@ -2395,3 +2398,298 @@ func Test_parseParentAndPlugin_RecursivelyReference_withMultipleURI(t *testing.T
+			curDevfile: DevfileObj{
+				Data: &v2.DevfileV2{
+					Devfile: v1.Devfile{
+						DevWorkspaceTemplateSpec: v1.DevWorkspaceTemplateSpec{
+							Parent: &v1.Parent{
+								ImportReference: v1.ImportReference{
+									ImportReferenceUnion: v1.ImportReferenceUnion{
+										Uri: localRelativeURI,
+									},
+								},
+							},
+							DevWorkspaceTemplateSpecContent: v1.DevWorkspaceTemplateSpecContent{},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "case 4: should fail if file not exist",
+			curDevfile: DevfileObj{
+				Ctx: devfileCtx.NewDevfileCtx(OutputDevfileYamlPath),
+				Data: &v2.DevfileV2{
+					Devfile: v1.Devfile{
+						DevWorkspaceTemplateSpec: v1.DevWorkspaceTemplateSpec{
+							Parent: &v1.Parent{
+								ImportReference: v1.ImportReference{
+									ImportReferenceUnion: v1.ImportReferenceUnion{
+										Uri: "notexist/devfile.yaml",
+									},
+								},
+							},
+							DevWorkspaceTemplateSpecContent: v1.DevWorkspaceTemplateSpecContent{},
+						},
+					},
+				},
+			},
+			uri:     "notexist/devfile.yaml",
+			wantErr: true,
+		},
+		{
+			name: "case 5: should fail if url not exist",
+			curDevfile: DevfileObj{
+				Ctx: devfileCtx.NewURLDevfileCtx(httpPrefix + uri1),
+				Data: &v2.DevfileV2{
+					Devfile: v1.Devfile{
+						DevWorkspaceTemplateSpec: v1.DevWorkspaceTemplateSpec{
+							Parent: &v1.Parent{
+								ImportReference: v1.ImportReference{
+									ImportReferenceUnion: v1.ImportReferenceUnion{
+										Uri: notExistURI,
+									},
+								},
+							},
+							DevWorkspaceTemplateSpecContent: v1.DevWorkspaceTemplateSpecContent{},
+						},
+					},
+				},
+			},
+			uri:     notExistURI,
+			wantErr: true,
+		},
+		{
+			name: "case 6: should fail if with invalid URI format",
+			curDevfile: DevfileObj{
+				Ctx: devfileCtx.NewURLDevfileCtx(OutputDevfileYamlPath),
+				Data: &v2.DevfileV2{
+					Devfile: v1.Devfile{
+						DevWorkspaceTemplateSpec: v1.DevWorkspaceTemplateSpec{
+							Parent: &v1.Parent{
+								ImportReference: v1.ImportReference{
+									ImportReferenceUnion: v1.ImportReferenceUnion{
+										Uri: invalidURL,
+									},
+								},
+							},
+							DevWorkspaceTemplateSpecContent: v1.DevWorkspaceTemplateSpecContent{},
+						},
+					},
+				},
+			},
+			uri:     invalidURL,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// if the main devfile is from local, need to set absolute path
+			if tt.curDevfile.Ctx.GetURL() == "" {
+				err := tt.curDevfile.Ctx.SetAbsPath()
+				if err != nil {
+					t.Errorf("Test_parseFromURI() unexpected error = %v", err)
+					return
+				}
+			}
+			got, err := parseFromURI(tt.uri, tt.curDevfile)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Test_parseFromURI() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if err != nil && tt.wantErr {
+				return
+			}
```

Comment:
lets try switching to this if condition, it handles everything `if tt.wantErr == (err == nil) {}`

Reply:
updated